interfaces/api/pipeline/IRelocalizationPipeline.h

```diff
@@ -23,12 +23,47 @@
 #include "datastructure/Image.h"
 #include "datastructure/Map.h"
 #include <xpcf/core/helpers.h>
-
+#include <core/SerializationDefinitions.h>
 
 namespace SolAR {
 namespace api {
 namespace pipeline {
 
+/**
+ * @typedef DetectedObjectType
+ * @brief <B>Define the types of objects that can be detected in an image.</B>
+ */
+typedef enum {
+    UNKNOWN_OBJECT = 0,
+    FIDUCIAL_MARKER_OBJECT = 1,
+    QR_CODE_OBJECT = 2,
+    NATURAL_IMAGE_OBJECT = 3
+} DetectedObjectType;
+
+/**
+ * @struct DetectedObject
+ * @brief <B>Define any object that can be detected in an image.</B>
+ */
+struct DetectedObject
```

Comment:
Move DetectedObject datastructure into its own file ? To avoid exposing serialization code in the pipeline interface ?

Reply:
Je crée un nouveau fichier dans "datastructure" ? Ca vous va ?

Reply:
Yes 👍